packages/cli/src/clean.ts

```diff
@@ -0,0 +1,281 @@
+            },
+            iosPath
+          );
+        }
+      );
+
+      break;
+    }
+    case "macos": {
+      findPath(cliOptions.projectRoot, "-macOS", (path: string) => {
+        const macosPath = path.substring(0, path.lastIndexOf("/"));
+
+        cleanDir({ name: "Pods Cache", path: `${macosPath}/Pods` });
+        cleanDir({ name: "CocoaPods cache", path: "~/.cocoapods" });
+        execute(
+          {
+            name: "Clean Pod Cache",
+            command: "pod",
+            args: ["cache", "clean", "--all"],
+          },
+          macosPath
+        );
+      });
+
+      break;
+    }
+  }
+}
+
+function findPath(
+  startPath: string,
+  filter: string,
+  callback: (path: string) => void
+) {
+  const files = fs.readdirSync(startPath);
+
+  for (let i = 0; i < files.length; i++) {
+    const filename = path.join(startPath, files[i]);
+
+    const stat = fs.lstatSync(filename);
+    if (stat.isDirectory()) {
+      //no need to check these dirs
+      if (
+        filename.indexOf(".git") >= 0 ||
+        filename.indexOf(".xcodeproj") >= 0 ||
+        filename.indexOf(".xcworkspace") >= 0 ||
+        filename.indexOf(".xcassets") >= 0
+      )
+        continue;
+
+      findPath(filename, filter, callback); //recurse
+    } else if (filename.indexOf(filter) >= 0) {
+      callback(startPath);
+      break;
+    }
+  }
+}
+
+function cleanDir(item: CachePath) {
+  console.log(chalk.cyan("info"), `Cleaning ${item.name}`);
+
+  if (!dirExists(item.path)) {
+    console.log(chalk.yellow("warning:"), `No such directory : ${item.path}`);
+
+    return;
+  }
+
+  fs.rmdirSync(item.path, { recursive: true });
+  console.log(chalk.cyan("info"), `Successfully cleaned: ${item.name}`);
+}
+
+function execute(item: CLICommand, rootPath: string) {
+  return new Promise((resolve, reject) => {
+    console.log(chalk.cyan("info"), item.name);
+    console.log(chalk.dim(`${item.command} ${item.args.join(" ")}`));
+
+    const { spawnSync } = require("child_process");
+
+    const task = spawnSync(item.command, item.args, {
+      cwd: rootPath ? path.resolve(rootPath) : undefined,
+      stdio: "inherit",
+    });
+
+    if (task.error && task.error.code == "ENOENT") {
+      console.log(
+        chalk.red.bold("Error"),
+        `${item.command} Failed! Command not found`
+      );
+      reject();
+    } else if (task.error) {
+      console.log(chalk.red.bold("Error"), task.error);
+      reject();
+    } else if (task.signal) {
+      console.log(
+        chalk.red.bold("Error"),
+        `Failed with signal ${task.signal}'`
+      );
+      reject();
+    } else if (task.status !== 0) {
+      console.log(
+        chalk.red.bold("Error"),
+        `Failed with exit code ${task.status}'`
+      );
+      reject();
+    }
+
+    console.log(`${item.name} completed successfully!`);
+    resolve(task.status);
+  });
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function QueryPromise(promise: any) {
```

Comment:
Please don't use `any`. Can we use generics instead?
```suggestion
function QueryPromise<T>(promise: Promise<T>) {
```